test/metabase/test/data/sql.clj

```diff
@@ -335,6 +335,33 @@
            query           (str/replace query (re-pattern #"WHERE .* = .*") (format "WHERE {{%s}}" (name field)))]
        {:query query}))))
 
+(defmethod tx/arbitrary-select-query :sql/test-extensions
+  ([driver table to-insert]
+   (driver/with-driver driver
+     (let [mbql-query      (data/mbql-query nil
+                             {:source-table (data/id table)
+                              :expressions  {:custom [:value 1337 {:base_type :type/Integer}]}
+                              :fields       [[:expression :custom]]
+                              :order-by     [[:asc [:field (data/id table :id)]]]
+                              :limit        3})
+           {:keys [query]} (qp.compile/compile mbql-query)
+           query           (str/replace query (re-pattern #"(.*)(?:1337)(.*)") (format "$1%s$2" to-insert))]
+       {:query query}))))
+
+(defmethod tx/field-reference :sql/test-extensions
+  ([driver field-id]
+   (->> [:field field-id {}]
+        (sql.qp/->honeysql driver)
+        (sql.qp/format-honeysql driver)
+        first)))
+
+(defmethod tx/field-reference :sparksql
+  ([driver field-id]
+   (let [parent-method (get-method tx/field-reference :sql/test-extensions)
+         full-reference (parent-method driver field-id)
+         [_ _ field-name] (str/split full-reference #"\.")]
+     (format "`t1`.%s" field-name))))
```

Comment:
huh, doesn't `->honeysql` produce a valid field reference? Is `t1` always the right prefix? I guess a comment here would be good.